http/go/nethttp/redirect_handler.go

```diff
@@ -0,0 +1,140 @@
+package nethttplibrary
+
+import (
+	"errors"
+	nethttp "net/http"
+	"net/url"
+	"strings"
+
+	abs "github.com/microsoft/kiota/abstractions/go"
+)
+
+type RedirectHandler struct {
+	options RedirectHandlerOptions
+}
+
+func NewRedirectHandler() *RedirectHandler {
+	return NewRedirectHandlerWithOptions(RedirectHandlerOptions{
+		MaxRedirects: DEFAULT_MAX_REDIRECTS,
+		ShouldRedirect: func(req *nethttp.Request, res *nethttp.Response) bool {
+			return true
+		},
+	})
+}
+func NewRedirectHandlerWithOptions(options RedirectHandlerOptions) *RedirectHandler {
+	return &RedirectHandler{options: options}
+}
+
+type RedirectHandlerOptions struct {
+	ShouldRedirect func(req *nethttp.Request, res *nethttp.Response) bool
+	MaxRedirects   int
+}
+
+var redirectKeyValue = abs.RequestOptionKey{
+	Key: "RedirectHandler",
+}
+
+type redirectHandlerOptionsInt interface {
+	abs.RequestOption
+	GetShouldRedirect() func(req *nethttp.Request, res *nethttp.Response) bool
+	GetMaxRedirect() int
+}
+
+func (o *RedirectHandlerOptions) GetKey() abs.RequestOptionKey {
+	return redirectKeyValue
+}
+func (o *RedirectHandlerOptions) GetShouldRedirect() func(req *nethttp.Request, res *nethttp.Response) bool {
+	return o.ShouldRedirect
+}
+func (o *RedirectHandlerOptions) GetMaxRedirect() int {
+	if o == nil || o.MaxRedirects < 1 {
+		return DEFAULT_MAX_REDIRECTS
+	} else if o.MaxRedirects > ABSOLUTE_MAX_REDIRECTS {
+		return ABSOLUTE_MAX_REDIRECTS
+	} else {
+		return o.MaxRedirects
+	}
+}
+
+var DEFAULT_MAX_REDIRECTS = 5
```

Comment:
Place constants in a different file?

Reply:
do you mean using the const keyword? it only works with string AFAIK

Reply:
Go supports constants of character, string, boolean, and numeric values, you can learn more [here](https://gobyexample.com/constants)

Reply:
Consider making these package private otherwise they'll leak to the users and you might save a few chars using mutliple initialization.
```go
const (
  absolute_max_redirects = 20
  ...
)
```

Reply:
thanks, I must have been confused with the key value which is a struct and cannot be a const.